django/core/checks/security/base.py

```diff
@@ -235,6 +246,38 @@ def check_referrer_policy(app_configs, **kwargs):
     return []
 
 
+@register(Tags.security, deploy=True)
+def check_cache_location(app_configs, **kwargs):
+    for cache_params in settings.CACHES.values():
+        if cache_params['BACKEND'] == 'django.core.cache.backends.filebased.FileBasedCache':
+            if settings.MEDIA_ROOT:
+                _media_root = os.path.abspath(settings.MEDIA_ROOT)
+                location_check = (
+                    _media_root,
+                    cache_params['LOCATION'],
+                )
+                if os.path.commonpath(location_check) == _media_root:
+                    return [W023]
+            if settings.STATIC_ROOT:
+                _static_root = os.path.abspath(settings.STATIC_ROOT)
+                location_check = (
+                    _static_root,
+                    cache_params['LOCATION'],
+                )
+                if os.path.commonpath(location_check) == _static_root:
+                    return [W023]
+            if settings.STATICFILES_DIRS:
```

Comment:
You should precompile a list of absolute paths, instead of having a loop within a loop.